alpa/device_mesh.py

```diff
@@ -154,60 +159,98 @@ def __init__(self, server_address: str, num_hosts: int, host_id: int,
         self.launched = True
 
     ##### Buffer Related Functions #####
-    def put_buffer(self, uuid: int, device_id: int, data: np.ndarray):
-        assert uuid not in self.buffers
-        if data.dtype == np.int64:
-            data = data.astype(np.int32)
-        self.buffers[uuid] = (self.backend.buffer_from_pyval(
-            data, self.local_devices[device_id]))
-
-    def put_buffers(self,
-                    uuids: Sequence[int],
-                    device_ids: Sequence[int],
-                    datas: Sequence[np.ndarray],
-                    num_batch=1,
-                    batch_dim=0):
+    def put_tensor(self,
```

Comment:
We have several names "buffer", "array", and "tensor". In the old codebase, we use "array" for a high-level logical array and "buffer" for a low-level per-device buffer. Jax codebase never uses "tensor".

In the new codebase, let me suggest a naming scheme:
1. Following the jax convention, never use "tensor"
2. Rename `RemoteBufferRef` -> `RemoteArrayRef`.
3. Do not change the name of `put_buffer`, `get_buffer` because they still work on low-level per-device buffers

Reply:
`put_buffer` cannot work on per-device buffers. `get_buffer` does so because there might be some redundancy due to the sharding spec, so we only get exactly one replica. currently `put_buffer` does not consider the exactly-one-replica stuff. We might do this optimization later so I reverted the name too.